pkg/counter/cache.go

```diff
@@ -219,6 +240,11 @@ func initialize(cl client.Client, toolchainStatus *toolchainv1alpha1.ToolchainSt
 		toolchainStatus.Status.HostOperator.MasterUserRecordCount == 0 {
 		return initializeFromResources(cl, toolchainStatus.Namespace)
 	}
+	// Migration for CRT-1072: force initialization from existing UserSignups/MasterUserRecords
+	// TODO: remove as part of CRT-1074
+	if _, exists := toolchainStatus.Status.Metrics[toolchainv1alpha1.MasterUserRecordsPerDomainMetricKey]; !exists {
+		return initializeFromResources(cl, toolchainStatus.Namespace)
+	}
```

Comment:
I'm curious - why should we remove it?
I would expect to be part of the previous if statement with the fact that it should stay there forever. 
https://github.com/codeready-toolchain/host-operator/blob/f21acd30e5bc453ddacd144b1c4f24cc936806d3/pkg/counter/cache.go#L239-L242
And we could do the same also for the other metric `toolchainv1alpha1.UsersPerActivationMetricKey`

Reply:
> I'm curious - why should we remove it?
> I would expect to be part of the previous if statement with the fact that it should stay there forever.
> https://github.com/codeready-toolchain/host-operator/blob/f21acd30e5bc453ddacd144b1c4f24cc936806d3/pkg/counter/cache.go#L239-L242
> 

good point. I guess I was in the mindset of the migration when I wrote this comment. I've updated the logic in 3b967e8

Reply:
> And we could do the same also for the other metric `toolchainv1alpha1.UsersPerActivationMetricKey`

done in 86045de

Reply:
I'm really sorry, but I got confused by the changes in the commit you referenced - you changed a lot of the code plus also the `setStateLabel` which is not related at all.
Are the changes in the tests related to the fact that it initializes the counter for every test? But why changing the `setStateLabel`?

Reply:
Now I see that the changes related to this specific metric were introduced in 3b967e8288765f71744bea25b94d392ce11c7bd3 - in the commit that you referenced, you added the `UsersPerActivationMetricKey` (which could have been done separately, but it probably doesn't matter now).

Reply:
ok, let me try to explain (or confirm what you wrote above):
- the changes in the `setStateLabel` func are just about renaming some local variables and adding comments. There is no change in the logic itself. I "touched" this func because I was confused with some tests results and while debugging, I found that the var deserved to be renamed to be more adequate with the new "state" stuff. That being said, I can revert the changes here.
- there are indeed a lot of changes related to the tests. As you noted, that's related to the counter initialization, which will look at the resources (`UserSignup` and `MasterUserRecord`) if one if the metric is missing. So, to avoid switching to that kind if initialization, I had to set the `UsersPerActivationMetricKey` metric to all these tests.

I hope I answered all your questions, so please let me know if you have further concerns